lxdprofile.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package charm
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/juju/utils/set"
+	"gopkg.in/yaml.v2"
+)
+
+type LXDProfileCharm interface {
+	Charm
```

Comment:
I don't think we should make the `LXDProfileCharm` a composition of `Charm` and `LXDProfiler`. With that I think this should be renamed to `LXDProfiler`.